public/app/features/alerting/unified/components/rules/deleted-rules/ConfirmRestoreDeletedRuleModal.tsx

```diff
@@ -0,0 +1,165 @@
+      confirmText={confirmText}
+      modalClass={styles.modal}
+      confirmButtonVariant={!error ? 'destructive' : 'primary'}
+      body={
+        <Stack direction="column" gap={2}>
+          <Trans i18nKey="alerting.deletedRules.restore-modal.body">
+            Are you sure you want to restore this deleted alert rule definition?
+          </Trans>
+
+          <div>{ruleToRestore && <RulePeview rule={ruleToRestore} />}</div>
+          {error && (
+            <Alert severity="warning" title={errorTitle}>
+              <Trans i18nKey="alerting.alertVersionHistory.restore-deleted-manually">
+                Your alert rule could not be restored. This may be due to changes to other entities such as contact
+                points, data sources etc. Please manually restore the deleted rule by editing the rule and saving it.
+              </Trans>
+              <pre style={{ marginBottom: 0 }}>
+                <code>{stringifyErrorLike(error)}</code>
+              </pre>
+            </Alert>
+          )}
+        </Stack>
+      }
+      onConfirm={!error ? onRestoreConfirm : onManualRestore}
+      onDismiss={onDismiss}
+    />
+  );
+};
+
+function RulePeview({ rule }: { rule: RulerRuleDTO }) {
+  const styles = useStyles2(getStyles);
+  return (
+    <div className={styles.content}>
+      <CodeEditor
+        width="100%"
+        height={600}
+        language={'json'}
+        value={JSON.stringify(rule, null, 4)}
+        monacoOptions={{
+          minimap: {
+            enabled: false,
+          },
+          scrollBeyondLastLine: false,
+          lineNumbers: 'on',
+          readOnly: true,
+        }}
+      />
+    </div>
+  );
+}
+
+const getStyles = () => ({
+  content: css({
+    flex: '1 1 100%',
+  }),
+  modal: css({
+    width: '700px',
+  }),
+});
+
+export function useRestoreDeletedRule() {
+  const [addRuleToRuleGroup] = useAddRuleToRuleGroup();
+
+  return useAsync(async (deletedRule: RulerGrafanaRuleDTO<GrafanaRuleDefinition>) => {
+    const ruleGroupIdentifier = {
+      dataSourceName: GRAFANA_RULES_SOURCE_NAME,
+      namespaceName: deletedRule.grafana_alert.namespace_uid,
+      groupName: deletedRule.grafana_alert.rule_group,
+    };
+    // save the new rule to the rule group
+    return addRuleToRuleGroup.execute(ruleGroupIdentifier, deletedRule);
+  });
+}
+
+const redirectToRestoreForm = async (ruleToRecover: RulerGrafanaRuleDTO) => {
+  let formValues: Partial<RuleFormValues> | undefined;
+  const namespaceName = await backendSrv
+    .getFolderByUid(ruleToRecover.grafana_alert.namespace_uid)
+    .then((folder) => folder.title);
+
+  try {
+    formValues = grafanaRuleDtoToFormValues(ruleToRecover, namespaceName);
+  } catch (err) {
+    const message = `Error getting rule values from the deleted rule: ${getMessageFromError(err)}`;
+    throw new Error(message);
+  }
+
+  const urlPath = rulerRuleType.any.recordingRule(ruleToRecover) ? '/alerting/new/grafana-recording' : '/alerting/new';
```

Comment:
Restoring is only supported for Grafana managed, right? Might be clearer to check `rulerRuleType.grafana.recording()` instead? As reading this implies that all recording rules go to the `grafana-recording` form

Reply:
agree